### PR TITLE
Fix: Add completion icon styles (fixes #99)

### LINKED ISSUE #99: Missing a visual completion state other than colour
Viewed/completed items indicate their completed state with a change in image or border colour. This information will be difficult or impossible to perceive for individuals with visual disabilities such as low vision or color blindness.

Provide a non-color dependent means of identifying viewed/completed items. For Hotgraphic, this was implemented with a checkmark icon over each pin to indicate the completed state - https://github.com/adaptlearning/adapt-contrib-hotgraphic/pull/231

### PR DESCRIPTION
**Pull request by swashbuck**
Fixes #99 

### Fix
* Adds a completion icon to the visited item state

### CODE REVIEW

**Comment by kirsty-hames** [comment]
We currently have a duplication of icons when items don't have titles set `_graphic.title`. See the first item in the screenshot below.
icon_duplication
Personally, I don't think we need the plus icon `.hotgrid__item-icon` and I find this regularly gets requested to be removed in projects anyway. So removing this would resolve the duplication.

File: less/hotgrid.less [hunk 1]
```diff
@@ -32,6 +37,25 @@
     .icon-plus;
   }
 
+  &__item-completion-icon {
+    display: none;
+    position: absolute;
+    bottom: 0.5rem;
+    right: 0.5rem;
+    border-radius: 50%;
+    padding: 1rem;
```
Comment (kirsty-hames) [inline less/hotgrid.less:46]: The icon overlaps the image which could be an issue for some imagery however reducing the padding will mean less of the image is obscured. I think 0.5rem looks good.

e.g `padding: @item-padding / 2`

**Comment by StuartNicholls** [comment]
I agree, I don't see the value of that plus icon. It doesn't really communicate that its clickable. its just similar to hotgraphic and I can't see the usefulness beyond that

File: less/hotgrid.less (hunk 1 above)
Comment (kirsty-hames) [reply less/hotgrid.less:46]: Also, I recall we were querying `.5rem` vs `0.5rem` on a recent PR (can't remember where I saw this). Do you remember if we agreed to either @swashbuck?

Comment (StuartNicholls) [reply less/hotgrid.less:46]: Personally, I prefer 0.5rem.

Comment (kirsty-hames) [reply less/hotgrid.less:46]: > Personally, I prefer 0.5rem.

My preference would be 0.5rem as my eyes sometimes miss the decimal 👀

Comment (swashbuck) [reply less/hotgrid.less:46]: @kirsty-hames Yeah, that was on the nav labels ticket. We ended up using an existing variable, but I agree that 0.5rem is less prone to misreading. All in favor of keeping the leading zero.

Comment (swashbuck) [reply less/hotgrid.less]: @kirsty-hames Ok, padding reduced to 0.5rem ( @item-padding / 2 )

**Comment by swashbuck** [comment]
> I agree, I don't see the value of that plus icon. It doesn't really communicate that its clickable. its just similar to hotgraphic and I can't see the usefulness beyond that.

@StuartNicholls @kirsty-hames Plus icon has been removed [e40316c](https://github.com/cgkineo/adapt-hotgrid/pull/117/commits/e40316c01b837eedaeb43c2fb635aa039f483467)

**Comment by oliverfoster** [comment]
This has scope creeped quite some. Please update the title+description or move the icon removal to a different pr. Perhaps the icon should just be configurable rather than removed? It's been there since 2019.

**Comment by StuartNicholls** [comment]
I do feel this was added in error, and as @kirsty-hames says, It's a bad fix to an issue that doesn't exist, but there are other things to discuss related to this so yep, @oliverfoster is right, lets put this in a separate PR. Thanks @swashbuck , hope that's okay

**Comment by swashbuck** [comment]
All good. Easy enough to revert [e40316c](https://github.com/cgkineo/adapt-hotgrid/pull/117/commits/e40316c01b837eedaeb43c2fb635aa039f483467). Should be good now.

**Comment by swashbuck** [comment]
Thanks for raising #119 @kirsty-hames .

Would it make sense to just move the completion icon to the upper left or right?

**Comment by kirsty-hames** [comment]
> Thanks for raising #119 @kirsty-hames .
> 
> Would it make sense to just move the completion icon to the upper left or right?

When items are irregular images or `_isRound` I think positioning the completion icon becomes tricky as the alignment with the image varies depending on the item width/responsiveness (see screenshots). Based on this, I think the positioning works best bottom center as per the existing `.hotgrid__item-icon` positioning. 

If you agree, I think we either remove `.hotgrid__item-icon` or go with your [suggestion in #199](https://github.com/cgkineo/adapt-hotgrid/issues/119#issuecomment-1535052393)?
round_items
round_items_mobile

**Comment by swashbuck** [comment]
@kirsty-hames I see what you mean about responsive layouts. If the completion icon is off to the side, it won't consistently be in the exact same position at all screen sizes.

On the other hand, moving the icon to bottom center doesn't address a previous concern about the icon obscuring part of the image and actually makes it worse.

So, this might be a case of the lesser of two evils: 1) inconsistent placement of the completion icon on responsive sizes or 2) having the icon obscure the image.

**Comment by kirsty-hames** [comment]
> @kirsty-hames I see what you mean about responsive layouts. If the completion icon is off to the side, it won't consistently be in the exact same position at all screen sizes.
> 
> On the other hand, moving the icon to bottom center doesn't address a previous concern about the icon obscuring part of the image and actually makes it worse.
> 
> So, this might be a case of the lesser of two evils: 1) inconsistent placement of the completion icon on responsive sizes or 2) having the icon obscure the image.

This makes sense thanks @swashbuck. Seeing as the `.hotgrid__item-icon` is already central I'd be inclined to center the completion icon. In the instance you have multiple Hotgrids within a course, if you have a mix of ones with titles and ones without, the icon is always positioned the same. Alternatively, if point 2 is the bigger issue then reposition both icons (e.g. bottom right)? I'd be happy with either if you have a preference.

**Comment by swashbuck** [comment]
@kirsty-hames I like the idea of moving the completion icon to the bottom center so that it is consistent with the `.hotgrid__item-icon` element. I don't believe point 2 is that big of a concern.

@StuartNicholls Any thoughts before I move the completion icon to bottom center?

**Comment by swashbuck** [comment]
Ok, I'm now using `.hotgrid__item-icon` for both the plus icon and checkmark icon. The plus icon will only be shown when there is no title (as it currently works). When visited, the checkmark icon will be displayed for all cases.

Also, I reduced the padding of this icon per [this comment](https://github.com/cgkineo/adapt-hotgrid/issues/119#issuecomment-1536289535).

**Review by kirsty-hames — APPROVED** [review]
Thanks @swashbuck. Looking good and works as expected.

**Pull request merged**